src/Plugin/search_api/datasource/SolrDocument.php

```diff
@@ -192,7 +192,7 @@ public function loadMultiple(array $ids) {
         ->addCondition('search_api_id', $ids, 'IN')
         ->execute()
         ->getResultItems();
-      foreach ($results as $id => $result) {
+      foreach ($results as $result) {
         $documents[$id] = $this->getSolrDocumentFactory()->create($result);
```

Comment:
`$id` is used here, but would no longer be set to anything (different)? It _might_ still have the last value of the `foreach` above (at line 187), but not varying in this loop means that `loadMultiple()` would only ever return a single item as it would be constantly rewriting `$documents[$id]` with different documents?

Might instead be required/desirable to reverse the mapping from https://github.com/mkalkbrenner/search_api_solr/pull/98/files#diff-ff43eafed5c89c92f4e2110430d91777900b31451bb3f235afd5baadc990b08cR187-R189 , to strip off the `solr_document/` prefix to return back to whatever bare IDs (as were passed in), prior to returning?

Reply:
Yeah I think you're right, there should be the reverse mapping or keep the original mapping sent in to the method. I'll play around with this today and see if I can come up with something better, because currently its relying on the previously set $id in the foreach loop above only truly works when there's a single item in the loop. It was just the quickest solution to my use case. Thanks for your input!

Reply:
So I'm just doing a str_replace now to remove the `solr_document/` portion of the $id. I think this makes more sense now, especially considering this is the $id coming into the method in the first place.